src/Sylius/Bundle/CoreBundle/Controller/WishlistController.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Controller;
+
+use Sylius\Bundle\CoreBundle\Doctrine\ORM\ProductRepository;
+use Sylius\Bundle\ResourceBundle\Controller\ResourceController;
+use Sylius\Component\Core\Model\WishlistInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class WishlistController extends ResourceController
+{
+    public function setAsPublicAction(Request $request, $id)
+    {
+        /** @var $wishlist WishlistInterface */
+        if (!$product = $this->getWishlistRepository()->findOneBy(array('id' => $id, 'user' => $this->getUser()))) {
```

Comment:
You have a double query here (next to the findOr404 later on) and the product variable is probably wrong.